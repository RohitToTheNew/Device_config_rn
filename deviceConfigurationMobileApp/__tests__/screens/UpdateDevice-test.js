import React from 'react';
import UpdateDevice from '../../src/screens/updateDevice/index';
import renderer from 'react-test-renderer';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {
  writeDataTolocalStorage,
  readFileFromLocalStorage,
  roomDropDown,
  updateSelectedSchool,
  updateSelectedRoom,
  schoolListUpdate,
  roomListUpdate,
  searchBarClose,
  onChangeTextFunction,
  searchFilter,
  deviceTypeDropDown,
  getRoom,
  updateDevice,
  onBackPressSaveDevice,
  navigateToDeviceListing,
  updateDeviceType,
} from '../../src/screens/updateDevice/action';

import Toast from 'react-native-toast-message';
import {NavigationContainer} from '@react-navigation/native';
import RNFS from 'react-native-fs';

const props = {
  writeDataTolocalStorage: jest
    .fn()
    .mockImplementation(() => Promise.resolve()),
  route: {params: []},
  navigation: {push: () => jest.fn(), goBack: () => jest.fn()},
};

describe('on landing to Update device', () => {
  let tree, navigationSpy;
  beforeEach(() => {
    navigationSpy = jest.spyOn(props.navigation, 'push');
    tree = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
    render(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
  });
  it('should render add device correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('Write file', () => {
  const mockData = [
    {
      id: 1,
      title: 'School 1',
      rooms: [
        {
          id: 1,
          title: 'Room 1 for School 1',
          devices: [
            {
              macAddress: 'macaddreeesssss12',
              serialNumber: 'serial number 2',
              deviceName: "Developer's device",
              deviceType: 'MS700',
              deviceID: 2,
            },
          ],
        },
      ],
    },
  ];

  test('should write a file', async () => {
    const result = await writeDataTolocalStorage(mockData);
    expect(result).toEqual('FILE WRITTEN!');
  });
  test('should throw the error if write fails', async () => {
    jest.spyOn(RNFS, 'writeFile').mockImplementation(() => {
      throw new Error('Error while writing');
    });

    const result = await writeDataTolocalStorage(mockData);
    expect(result).toEqual('Error while writing');
  });
});

describe('deviceTypeDropDown pressed', () => {
  it('model should be visible', () => {
    let component = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );

    const deviceTypeDropDown = component.root.findByProps({
      testID: 'deviceTypeDropDown',
    }).props;
    deviceTypeDropDown.onPress();

    const dropDownHeading = component.root.findByProps({
      testID: 'dropDownHeading',
    }).props.children;

    expect(dropDownHeading).toBe('Select Device Type');
  });
});

describe('schoolDropDown pressed', () => {
  it('model should be visible', () => {
    let component = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );

    const schoolDropDown = component.root.findByProps({
      testID: 'schoolDropDownbtn',
    }).props;
    schoolDropDown.onPress();

    const dropDownHeading = component.root.findByProps({
      testID: 'dropDownHeading',
    }).props.children;
    expect(dropDownHeading).toBe('Select School');
  });
});

describe('on selecting room dropdown', () => {
  beforeEach(() => {
    component = (
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>
    );
    render(component);
  });

  test('if school is not selected should show toast', () => {
    let schoolId = 0;
    const setIsRoomDropDown = () => {};
    const setDeviceTypeSelected = () => {};
    const setModalVisible = () => {};
    const modalVisible = () => {};
    const setSearch = () => {};
    const setRoomData = () => {};
    const setDataSource = () => {};

    const toastSpy = jest.spyOn(Toast, 'show');
    roomDropDown(
      schoolId,
      setIsRoomDropDown,
      setDeviceTypeSelected,
      setModalVisible,
      modalVisible,
      setSearch,
      setRoomData,
      setDataSource,
    );
    expect(toastSpy).toBeCalled();
  });

  test('if school is selected, model should open', () => {
    let schoolId = 1;
    const setIsRoomDropDown = () => {};
    const setDeviceTypeSelected = () => {};
    const setModalVisible = () => {};
    const modalVisible = () => {};
    const setSearch = () => {};
    const setRoomData = () => {};
    const setDataSource = () => {};
    roomDropDown(
      schoolId,
      setIsRoomDropDown,
      setDeviceTypeSelected,
      setModalVisible,
      modalVisible,
      setSearch,
      setRoomData,
      setDataSource,
    );
    let component = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
    const dropDownHeading = component.root.findByProps({
      testID: 'dropDownHeading',
    }).props.children;

    expect(dropDownHeading).toBe('Select School');
  });
});

describe('updateSelectedSchool called', () => {
  const dataSource = [
    {
      id: 1,
      title: 'School 1',
      rooms: [
        {
          id: 1,
          title: 'Room 1',
          devices: [
            {
              macAddress: 'AAAAAAAAAAAAAAAAA',
              serialNumber: 'serial number',
            },
          ],
        },
      ],
    },
  ];
  const title = {title: {item: {title: 'School 1'}}};
  const setSchoolId = () => {};
  const setSelectedSchool = () => {};
  const setSelectedRoom = () => {};
  const setModalVisible = () => {};
  const setSearch = () => {};
  const setSchoolSelectedFlag = () => {};

  updateSelectedSchool(
    dataSource,
    title,
    setSchoolId,
    setSelectedSchool,
    setSelectedRoom,
    setModalVisible,
    setSearch,
    setSchoolSelectedFlag,
  );

  let component = renderer.create(
    <NavigationContainer>
      <UpdateDevice {...props} />
    </NavigationContainer>,
  );

  test('model should be close after school is selected', () => {
    const modelComponent = component
      .toJSON()
      .children.filter(data => data.type === 'Modal');
    const modelForListComponent = modelComponent.filter(
      data => data.props.testID === 'modelForList',
    );
    expect(modelForListComponent?.[0].props.visible).toBeFalsy();
  });
});

describe('Create new School', () => {
  const schoolData = [
    {
      id: 1,
      title: 'School 1',
      rooms: [
        {
          id: 1,
          title: 'Room 1 for School 1',
          devices: [
            {
              macAddress: 'macaddreeesssss12',
              serialNumber: 'serial number 2',
              deviceName: "Developer's device",
              deviceType: 'MS700',
              deviceID: 2,
            },
          ],
        },
      ],
    },
  ];
  const setModalVisible = () => {};
  const setSearch = () => {};
  const setDataSource = () => {};
  const setSchoolData = () => {};
  const dataSource = [
    {
      id: 1,
      title: 'School 1',
      rooms: [
        {
          id: 1,
          title: 'Room 1 for School 1',
          devices: [
            {
              macAddress: 'macaddreeesssss12',
              serialNumber: 'serial number 2',
              deviceName: "Developer's device",
              deviceType: 'MS700',
              deviceID: 2,
            },
          ],
        },
      ],
    },
  ];
  const setSchoolId = () => {};
  const setSelectedSchool = () => {};
  const setSelectedRoom = () => {};
  const setSchoolSelectedFlag = () => {};

  schoolListUpdate(
    schoolData,
    setModalVisible,
    setSearch,
    setDataSource,
    setSchoolData,
    dataSource,
    setSchoolId,
    setSelectedSchool,
    setSelectedRoom,
    setSchoolSelectedFlag,
  )('new School');
});

describe('updateDevice', () => {
  let navigationSpy = '';
  let updatedDevice = {
    deviceID: '1',
    deviceName: 'dummy Device',
    deviceType: 'MS700',
    macAddress: 'MacAddressssss',
    serialNumber: 'Serial number 1',
  };

  beforeEach(() => {
    navigationSpy = jest.spyOn(props.navigation, 'goBack');
    tree = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
    render(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
  });

  test('it should navigate back after saving device', () => {
    updateDevice(updatedDevice);
    setTimeout(() => {
      expect(navigationSpy).toBeCalled();
    }, 100);
  });
});

describe('onBackPressSaveDevice', () => {
  let navigationSpy = '';
  const macAddress = 'AB:BC:SD:FR:FR:QW';
  const serialNumber = 'Serial number';
  const schoolId = 1;
  const roomId = 1;

  beforeEach(() => {
    navigationSpy = jest.spyOn(props.navigation, 'goBack');
    tree = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
    render(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
  });

  test('it should navigate back', () => {
    onBackPressSaveDevice(macAddress, serialNumber, schoolId, roomId);
    setTimeout(() => {
      expect(navigationSpy).toBeCalled();
    }, 100);
  });
});

describe('navigateToDeviceListing', () => {
  let navigationSpy = '';
  const macAddress = 'AB:BC:SD:FR:FR:QW';
  const serialNumber = 'Serial number';
  const schoolId = 1;
  const roomId = 1;

  beforeEach(() => {
    navigationSpy = jest.spyOn(props.navigation, 'push');
    tree = renderer.create(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
    render(
      <NavigationContainer>
        <UpdateDevice {...props} />
      </NavigationContainer>,
    );
  });

  test('it should navigate to manageSchool screen', () => {
    navigateToDeviceListing(macAddress, serialNumber, schoolId, roomId);
    setTimeout(() => {
      expect(navigationSpy).toBeCalled();
    }, 100);
  });
});

describe('Device type update', () => {
  const deviceType = 'MS700';
  const setDeviceType = () => {};
  const modalVisible = true;
  const setModalVisible = () => {};

  updateDeviceType(deviceType, setDeviceType, modalVisible, setModalVisible);

  let component = renderer.create(
    <NavigationContainer>
      <UpdateDevice {...props} />
    </NavigationContainer>,
  );

  test('model should be visible', () => {
    const modelForList = component.root.findByProps({
      testID: 'modelForList',
    }).props.children;
    expect(modelForList).toBe('Select School');
  });
});
