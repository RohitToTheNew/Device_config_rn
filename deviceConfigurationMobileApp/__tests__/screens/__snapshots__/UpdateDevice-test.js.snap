// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on landing to Update device should render add device correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#E5E5E5",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 32,
        "marginTop": 46,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <
        height={48}
        style={
          {
            "marginEnd": 16,
          }
        }
        width={48}
      />
    </View>
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#FB8C00",
          "flex": 1,
          "fontFamily": "CenturyGothic",
          "fontSize": 36,
          "fontWeight": "700",
        }
      }
    >
      Edit 
    </Text>
    <View />
    <Modal
      animationType="fade"
      onRequestClose={[Function]}
      transparent={true}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignSelf": "flex-end",
              "backgroundColor": "white",
              "borderRadius": 16,
              "elevation": 4,
              "justifyContent": "flex-start",
              "marginHorizontal": 60,
              "marginTop": 150,
              "minWidth": 340,
              "paddingEnd": 32,
              "paddingStart": 24,
              "paddingVertical": 24,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 4,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 24,
                "opacity": 1,
              }
            }
          >
            <
              height={40}
              width={40}
            />
            <Text
              style={
                {
                  "color": "#1A1C1C",
                  "fontFamily": "CenturyGothic-Bold",
                  "fontSize": 28,
                  "marginBottom": 4,
                  "marginLeft": 32,
                  "marginTop": 2,
                }
              }
            >
              [missing "en.identify" translation]
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 24,
                "opacity": 1,
              }
            }
          >
            <
              height={40}
              width={40}
            />
            <Text
              style={
                {
                  "color": "#1A1C1C",
                  "fontFamily": "CenturyGothic-Bold",
                  "fontSize": 28,
                  "marginBottom": 4,
                  "marginLeft": 32,
                  "marginTop": 2,
                }
              }
            >
              [missing "en.resetPasscode" translation]
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "padding": 15,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 60,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#808284",
            "flex": 1,
            "fontFamily": "CenturyGothic",
            "fontSize": 20,
            "paddingLeft": 0,
            "paddingTop": 12,
          }
        }
      >
        [missing "en.selectSchoolOrTypeToAddNew" translation]
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
            testID="schoolDropDownbtn"
          >
            [missing "en.selectSchool" translation]
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flex": 0.1,
              "marginRight": 5,
            }
          }
        >
          <
            height={30}
            width={30}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 60,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#808284",
            "flex": 1,
            "fontFamily": "CenturyGothic",
            "fontSize": 20,
            "paddingLeft": 0,
            "paddingTop": 12,
          }
        }
      >
        [missing "en.selectRoomOrTypeToAddNew" translation]
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
            testID="roomDropDownbtn"
          >
            [missing "en.selectRoom" translation]
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flex": 0.1,
              "marginRight": 5,
            }
          }
        >
          <
            height={30}
            width={30}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 60,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#808284",
            "flex": 1,
            "fontFamily": "CenturyGothic",
            "fontSize": 20,
            "paddingLeft": 0,
            "paddingTop": 12,
          }
        }
      >
        [missing "en.selectDeviceTypeOrTapToAddNew" translation]
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
            testID="deviceTypeDropDown"
          />
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flex": 0.1,
              "marginRight": 5,
            }
          }
        >
          <
            height={30}
            width={30}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 60,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#808284",
            "flex": 1,
            "fontFamily": "CenturyGothic",
            "fontSize": 20,
            "paddingLeft": 0,
            "paddingTop": 12,
          }
        }
      >
        [missing "en.selectDeviceNameOrAddNew" translation]
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <TextInput
            maxLength={45}
            onChangeText={[Function]}
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 60,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#808284",
            "flex": 1,
            "fontFamily": "CenturyGothic",
            "fontSize": 20,
            "paddingLeft": 0,
            "paddingTop": 12,
          }
        }
      >
        [missing "en.macAddress" translation]
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <TextInput
            maxLength={17}
            onChangeText={[Function]}
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 60,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#808284",
            "flex": 1,
            "fontFamily": "CenturyGothic",
            "fontSize": 20,
            "paddingLeft": 0,
            "paddingTop": 12,
          }
        }
      >
        [missing "en.serialNumber" translation]
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginTop": 100,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E53935",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
            "width": "48%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "CenturyGothic",
              "fontSize": 32,
              "fontWeight": "700",
            }
          }
        >
          Delete
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#808284",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
            "paddingRight": 5,
            "width": "48%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "CenturyGothic",
              "fontSize": 32,
              "fontWeight": "700",
            }
          }
        >
          Update
        </Text>
      </View>
    </View>
  </View>
</View>
`;
